src/shared/hooks/contracts/collective-rewards/useReadGauges.ts

```diff
@@ -0,0 +1,65 @@
+import { getAbi, type GaugeAbi } from '@/lib/abis/v2'
+import { AVERAGE_BLOCKTIME } from '@/lib/constants'
+import { useMemo } from 'react'
+import { Abi } from 'viem'
+import { UseReadContractParameters, UseReadContractReturnType, useReadContracts } from 'wagmi'
+import { UseReadContractsConfig, ViewPureFunctionName } from '../types'
+
+type GaugeFunctionName = ViewPureFunctionName<GaugeAbi>
+
+export type ReadGaugesConfig<TFunctionName extends GaugeFunctionName> = UseReadContractsConfig<
+  GaugeAbi,
+  TFunctionName
+>
+
+type ReadGaugesReturnType<
+  TAbi extends Abi,
+  TFunctionName extends ViewPureFunctionName<TAbi>,
+> = UseReadContractReturnType<TAbi, TFunctionName>['data']
+
+const abi = getAbi('GaugeAbi')
+
+export const useReadGauges = <TFunctionName extends GaugeFunctionName>(
+  { addresses, ...config }: ReadGaugesConfig<TFunctionName>,
+  query?: Omit<UseReadContractParameters<GaugeAbi, TFunctionName>['query'], 'select'>,
+) => {
+  const { data: results, ...queryData } = useReadContracts({
```

Comment:
I couldn't find in the doc if an error is raised even if only one request fails. Have we had the chance to check it?